src/main/java/ee/tuleva/onboarding/comparisons/fundvalue/retrieval/GlobalStockIndexCreator.java

```diff
@@ -0,0 +1,95 @@
+package ee.tuleva.onboarding.comparisons.fundvalue.retrieval;
+
+import ee.tuleva.onboarding.comparisons.fundvalue.FundValue;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Slf4j
+@Service
+public class GlobalStockIndexCreator implements ComparisonIndexRetriever {
```

Comment:
If this class implements the `ComparisonIndexRetriever` interface then it IS-A `Retriever`. Therefore, it should be named accordingly - i.e. `GlobalStockIndexRetriever` or something along those lines.

https://stackoverflow.com/questions/36162714/what-is-the-difference-between-is-a-relationship-and-has-a-relationship-in-java